src/lib/rest/RestService.cpp

```diff
@@ -442,8 +442,7 @@ std::string restService(ConnectionInfo* ciP, RestService* serviceV)
     // A tenant string must not be longer than 50 characters and may only contain
     // underscores and alphanumeric characters.
     //
-    std::string result;
-    if ((ciP->tenant != "") && ((result = tenantCheck(ciP->tenant)) != "OK"))
+    if ((ciP->tenant != "") && (tenantCheck(ciP->tenant)) != "OK")
     {
       OrionError  error(SccBadRequest,
                         "tenant name not accepted - a tenant string must not be longer than " MAX_TENANT_NAME_LEN_STRING " characters"
```

Comment:
This here seems like a mistake.
"result" should have been used, and not a fixed string (as detail for OrionError).
I just looked at tenantCheck() and (as I suspected) it has two ways of failing (invalid char + name too long).
Let's use this knowledge and fix this error, returning the correct error string (result).